pkg/trace/api/api.go

```diff
@@ -516,13 +516,18 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		atomic.AddInt64(&ts.PayloadRefused, 1)
 		return
 	}
+	if err != nil {
+		log.Errorf("Failed to count traces: %s", err)
+	}
 
 	start := time.Now()
-	defer func(err error) {
+	defer func() {
 		tags := append(ts.AsTags(), fmt.Sprintf("success:%v", err == nil))
 		metrics.Histogram("datadog.trace_agent.receiver.serve_traces_ms", float64(time.Since(start))/float64(time.Millisecond), tags, 1)
-	}(err)
-	tp, ranHook, err := decodeTracerPayload(v, req, ts)
+	}()
+	var tp *pb.TracerPayload
+	var ranHook bool
```

Comment:
This is a good change, but can you group these declarations together with parenthesis please?